.github/workflows/python-app.yml

```diff
@@ -151,3 +151,47 @@ jobs:
     - name: MyPy
       run: |
         mypy usl_pipeline
+
+  usl_models:
+    name: USL Models CI
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.11"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        cd usl_models
+        pip install -r requirements.txt
+        pip install -e .[dev]
+    - name: Lint with flake8
+      run: |
+        flake8 usl_models --show-source --statistics
+    - name: Ensure black auto-formatter has run
+      run: |
+        black usl_models --check
+    - name: Test with pytest
+      run: |
+        pytest usl_models
+
+  usl_models_type_checking:
+    name: USL Models Type Check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.11"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        cd usl_models
+        pip install -r requirements.txt
+        pip install -e .[dev]
+    - name: MyPy
+      run: |
+        mypy usl_models
```

Comment:
I'd just add the `mypy usl_models` step to the `USL Models CI` step above.
The reason I have it as separate above is to type check all the different usl_pipeline packages together. MyPy isn't as smart if it doesn't see the actual source files for imported code.

Reply:
Done.